test/customer.js

```diff
@@ -64,145 +64,270 @@ describe('Customer API:', () => {
+        whenDone(() => {
+          expect(requests.length).to.equal(1);
+          const req = requests[0];
+          expect(req.method).to.equal('POST');
+          expect(req.url).to.equal(
+            `${apiUrl}/workspaces/${config.workspaceId}/customers`
+          );
+          expect(JSON.parse(req.requestBody)).to.eql({
+            nodeId: config.nodeId,
+            externalId: mario.externalId,
+            base: mario.base
+          });
+          expect(req.requestHeaders.Authorization).to.equal(
+            `Bearer ${config.token}`
+          );
+          done();
+        });
+      });
+
+      it('handles 409 conflicts and updates the existing customer', (done) => {
+        whenDone(() => {
+          requests[0].respond(409, {}, JSON.stringify({
+            data: {
+              customer: {
+                id: 'existing-cid'
+              }
+            }
+          }));
+          whenDone(() => {
+            expect(requests.length).to.equal(2);
+            const req = requests[1];
+            expect(req.method).to.equal('PATCH');
+            expect(req.url).to.equal(
+              `${apiUrl}/workspaces/${config.workspaceId}/customers/existing-cid`
+            );
+            expect(JSON.parse(req.requestBody)).to.eql({
+              externalId: mario.externalId,
+              base: mario.base
+            });
+            done();
+          });
+        });
+      });
+
+      it('stores the customerId for future calls', (done) => {
+        whenDone(() => {
+          requests[0].respond(200, {}, JSON.stringify({ id: 'new-cid' }));
+          whenDone(() => {
+            expect(getCookie().customerId).to.equal('new-cid');
+            done();
+          });
+        });
+      });
+
+      it('stores a hash of the customer data for future calls', (done) => {
+        whenDone(() => {
+          requests[0].respond(200, {}, JSON.stringify({ id: 'new-cid' }));
+          whenDone(() => {
+            expect(getCookie().hash).not.to.be.undefined;
+            done();
+          });
+        });
+      });
+
+      it('reconciles the sessionId with the customerId', (done) => {
+        const sid = getCookie().sid;
+        whenDone(() => {
+          requests[0].respond(200, {}, JSON.stringify({ id: 'new-cid' }));
+          whenDone(() => {
+            expect(requests.length).to.equal(2);
+            const req = requests[1];
+            expect(req.url).to.equal(
+              `${apiUrl}/workspaces/${config.workspaceId}/customers/new-cid/sessions`
+            );
+            expect(JSON.parse(req.requestBody)).to.eql({
+              value: sid
+            });
+            done();
+          });
+        });
+      });
     });
 
-    it('creates a new customer', (done) => {
-      whenDone(() => {
+    describe('and a customerId is stored in the cookie', () => {
+      beforeEach(() => {
+        setConfig();
+        cookies.set(cookieName, Object.assign(getCookie(), {
```

Comment:
`store` is just one of many different calls of `cookies.set`. It is true that it is reused in a few places, but there are other places where `cookies.set` is called with different parameters. `cookies.set` already comes from an external library so I think it provides a reasonable level of abstraction.

I'm up for discussing a refactor of all cookie-related lines in a separate issue/PR.